scripts/pr_dev_version.js

```diff
@@ -0,0 +1,32 @@
+const fs = require('fs');
+
+function init() {
+  const packageJson = fs.readFileSync('./package.json', 'utf8');
+  const { version } = JSON.parse(packageJson);
+
+  // TODO ideally I wanted to check diffs against "main",
+  // but I don't know how to do it.
+  if (version.includes('-dev')) {
+    console.log(
+      '🟠 This PR cannot be merged because the package.json version contains ".dev-". ' +
+        '\n   Please change the version back to the original one. ' +
+        '\n   Make sure the version is exactly the same as in the main branch.'
+    );
+    process.exit(1);
+  }
```

Comment:
Depending on what you want to check in terms of diffs, you might be able to use `git diff` with some flags to achieve what you want.

For example, you can do:
```
git diff --name-status --merge-base main HEAD
```
which will show what happened to the file + the file name, something like this:
```
M       mix.exs
D       priv/json_schemas/common/contract_details/order_all.json
M       priv/json_schemas/contract_details/AUS.json
A       priv/json_schemas/contract_details/AUT.json
```
It essentially shows what changed between the current HEAD and the commit that's the common ancestor of the current branch and `main` (assuming there's only a single common ancestor, which should be the case for us)

Reply:
Today with fresh eyes I found a (different) way! Instead of using `git diff` to get the difference, I used `git show` to output the file content of main, and then parse it and manually compare it to the current branch. Check it out at https://github.com/remoteoss/json-schema-form/pull/3/commits/20b69f27eb14dcba86698766315a9538f9c8ae16

Reply:
Ah, although this worked locally, it [fails in CI](https://github.com/remoteoss/json-schema-form/actions/runs/5012270770/jobs/8984036926?pr=3) because it can't perform `git` commands. So I reverted the `pr_dev_version.js` to what I had before https://github.com/remoteoss/json-schema-form/pull/3/commits/241dfc2a37c966cbb7b5022f65b08ed2b4820716.

Reply:
Okay, now is working file, with [CI throwing the expected error](https://github.com/remoteoss/json-schema-form/actions/runs/5012509498/jobs/8984543336?pr=3). I've pushed a last commit to revert it back to `0.0.0` using `npm run version_as_main`. https://github.com/remoteoss/json-schema-form/pull/3/commits/241dfc2a37c966cbb7b5022f65b08ed2b4820716